### PR TITLE
Simplify display list construction in the API

### PR DESCRIPTION
**Pull request by mrobinson**
This makes it a bit less complicated to use the WebRender API and hides
some details that might change later.

---
This change is [Reviewable](https://reviewable.io/reviews/servo/webrender/592)

### CODE REVIEW

File: sample/src/main.rs [hunk 1]
```diff
@@ -132,10 +128,7 @@ fn main() {
             Rect::new(Point2D::new(50.0, 50.0), Size2D::new(100.0, 100.0)),
             radius);
 
-        webrender_traits::ClipRegion::new(&bounds,
-                                          vec![complex],
-                                          Some(mask),
-                                          &mut auxiliary_lists_builder)
+        builder.new_clip_region(&bounds, vec![complex], Some(mask))
```
Comment (kvark) [inline sample/src/main.rs]: that looks much nicer

File: webrender_traits/src/display_list.rs [hunk 2]
```diff
@@ -47,13 +48,15 @@ impl BuiltDisplayList {
 pub struct DisplayListBuilder {
     pub mode: DisplayListMode,
     pub list: Vec<DisplayItem>,
+    pub auxiliary_lists_builder: AuxiliaryListsBuilder,
```
Comment (kvark) [inline webrender_traits/src/display_list.rs]: does it need to be public?

**Comment by glennw** [comment]
r=me when rebased and servo patch is ready

File: webrender_traits/src/display_list.rs (hunk 2 above)
Comment (mrobinson) [inline webrender_traits/src/display_list.rs]: Nope! I will fix this

**Comment by mrobinson** [comment]
@glennw I have added a new commit to this PR. Do you mind taking a look at it and reviewing again?

The new commit: https://github.com/servo/webrender/pull/592/commits/5be4382515dfbdf0e393ab41b45ef94c6fcead45
The Servo portion: https://github.com/mrobinson/servo/tree/display-list-builder-updates

The Servo PR is waiting on the WebRender version update in order to update the lock files. Thanks!

**Comment by glennw** [comment]
@mrobinson Both the new commit and the servo PR look good to - feel free to merge this when the Servo PR is available.

**Comment by mrobinson** [comment]
@bors-servo r=glennw

@glennw Thanks for the review!

**Pull request merged**